### PR TITLE
Ampers Mariko Anderson

### PR DESCRIPTION
**Pull request by marikoja**
# Ride Share
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | While working on this project I had to choose how to account for in-progress trips while calculating driver revenue. I chose to loop over all trips, including the in-progress trips, and only change the revenue when there was an end of trip.
Describe any examples of composition that you encountered in this project, if any | in Ride Share the  a trip has a passenger and it has a driver. 
Describe a _nominal test_ that you wrote for this assignment. | One nominal test checks that the driver's trips return an array. 
Describe an _edge case test_ that you wrote for this assignment. | An edge case tests for a ride request when there are no drivers available. 
Describe a concept that you gained more clarity on as you worked on this assignment. | This assignment has helped me feel more confident in my test writing and the arrange portion of the tests.

### CODE REVIEW

**Comment by tildeee** [comment]
# Ride Share
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Used Git Regularly | Your commits tend to be on the larger side-- you can probably make them smaller!
Answer comprehension questions | x
**Wave 1** |
`Trip` has a helper method to calculate duration | x
`Passenger` has a method to calculate total cost of all trips | x
`Driver` has a method to calculate total revenue | x
Complete tests | Driver's revenue tests are incomplete all around (average revenue per hour only tests it's a float?), Passenger tests are only nominal cases
**Wave 2** |
`TripDispatcher` has a new method to create trips | x
Creating a trip in `TripDispatcher` relies on methods in `Driver` and `Passenger` to modify their own attributes | x
Methods from wave 1 work on `Driver`s and `Passenger`s with incomplete trips | x
Complete tests | x
**Wave 3** (optional) |
Complex logic was correctly implemented and tested | 
Appropriate use of Ruby's `Date` | 
Appropriate helper methods were made to help with complex logic | 
Complete tests | 
**Overall** | 

Overall, great job!

Great job practicing private methods, like in `Trip`

You tended to use `.each` iteration to calculate totals/sum things in Driver and Passenger, even though things like `.sum` is available. Honestly, I don't believe that you ALWAYS need to use fancy Enumerable methods like `.sum` over `.each` iteration every time, but I want to make sure you know that those are possibilities.

You actually put some of the edge cases I'm looking for (trips with `nil` end date) in the TripDispatcher spec in the `accept_trip method` test. I'm really glad you tested those things, like "does not update the passenger's total spent if end time is nil", but I could have also seen that test in just the passenger_spec.rb


Anyway, the code was pretty good! I'm glad you're feeling more comfortable with tests; it shows! I think there's a little bit more work to get to truly feeling good about tests, but it shows that you wrote testable code. Good work

File: specs/trip_spec.rb [hunk 1]
```diff
@@ -38,5 +41,16 @@
...
+
+    it "returns trip duration" do
+      @trip.trip_duration.must_be_kind_of Integer
+      @trip.trip_duration.must_equal 25
+    end
+
+    it "raises an ArgumentError" do
+      proc {
+        @trip_data[:end_time] = '2015-05-20T12:00:00+00:00'
+        RideShare::Trip.new(@trip_data).trip_duration}.must_raise ArgumentError
```
Comment (tildeee) [inline specs/trip_spec.rb:53]: This test just says that the method "should raise an ArgumentError," but doesn't explain why, and I can't tell just from this code _why_ this method raises one (especially from a business perspective). Maybe add to the it description "raises an ArgumentError _when end time is ..._" or use a local variable name to describe what is going on (invalid_time?)

That should push this test from testing something very important, to testing something very important *and* making it clear what's happening
